hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1540,13 +1510,18 @@ private HoodieData<HoodieRecord> getRecordIndexUpdates(HoodieData<WriteStatus> w
                   recordDelegate.getRecordKey(), recordDelegate.getPartitionPath(),
                   newLocation.get().getFileId(), newLocation.get().getInstantTime(), dataWriteConfig.getWritesFileIdEncoding());
             }
+            hoodieRecord = HoodieMetadataPayload.createRecordIndexUpdate(
```

Comment:
this does not look right. 
we should not add a entry to RLI metadata partition if it is an update record in data table. 
as per master, we will ignore it. but here looks like we are updating RLI even for update record. 
can you double check the logic once.